cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -165,8 +165,9 @@ FORCE_INLINE
 char* castVARCHAR_utf8_int64(int64 context, const char* data, int32 data_len,
                              int64_t out_len, int32_t* out_length) {
   // TODO: handle allocation failures
-  int32_t len = data_len <= static_cast<int32_t>(out_len) ? data_len
-                                                          : static_cast<int32_t>(out_len);
+  int32_t len = data_len <= static_cast<int32_t>(out_len) || out_len == 0
```

Comment:
What is the significance of this change? It seems unrelated to the concat functions and you haven't added a new test for `out_len == 0`

Reply:
this is not related to concat directly. I will create a new issue for this.